modules/search/sanctionsearch.py

```diff
@@ -0,0 +1,82 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+meta = {
+        'name': 'SanctionSearch',
+        'author': 'Kaushik',
+        'version': '0.1',
+        'description': 'SanctionSearch is an international list of blacklisted, dangerous\
+		personnel and entities.',
+        'sources': ('sanctionsearch'),
+        'options': (
+		('query', None, False, 'Name to search', '-q', 'store', str),
+		('id', None, False, 'Id to search', '-i', 'store', int),
+		('limit', 15, False, 'Max result count (default=15)', '-l', 'store', int),
+        ),
+        'examples': ('sanctionsearch -q <QUERY> -l 20',
+		'sanctionsearch -i <ID>')
+}
+
+NAMESEARCH = False
+
+def module_api(self):
+	global NAMESEARCH
+
+	query = self.options['query']
+	limit = self.options['limit']
+	id = self.options['id']
+
+	if query is None and id is None:
+		self.error('Either query or id is required')
+		return
+	elif query is not None and id is not None:
+		self.error('Only specify either query or id not both')
+		return
+	elif query is not None:
+		NAMESEARCH = True
+	
+	run = self.sanctionsearch(query=query, id=id, limit=limit)
+
+	if NAMESEARCH:
+		output_param = query
+		run.name_crawl()
+		output = {'results': []}
+		links = run.data
+		for item in links:
+			output['results'].append(item)
```

Comment:
You can use `extend`

Reply:
Each item is a dict of a row from the site, so they must be compartmentalized. Perhaps i modified an existing module and so the links variable name was left unchanged, I'll change that